src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -131,10 +132,11 @@ class SourcesTree extends Component<Props, State> {
       nextProps.selectedSource &&
       nextProps.selectedSource != selectedSource
     ) {
-      const highlightItems = getDirectories(
-        nextProps.selectedSource.get("url"),
-        sourceTree
-      );
+      let url = nextProps.selectedSource.get("url");
+      if (url) {
+        url = getRawSourceURL(url);
```

Comment:
Some `url`'s are `null`, so this `if` check needs to be done.

Reply:
Should we add a comment here? it might be hard to figure out why this raw source was required.

Reply:
hmm, what about making getRawSourceURL safe and return null if there is no URL?